dask/dataframe/shuffle.py

```diff
@@ -407,19 +412,26 @@ def rearrange_by_column_disk(df, column, npartitions=None, compute=False):
     dsk3 = {barrier_token: (barrier, list(dsk2))}
 
     # Collect groups
-    name = "shuffle-collect-" + token
+    name1 = "shuffle-collect-1" + token
     dsk4 = {
-        (name, i): (collect, p, i, df._meta, barrier_token) for i in range(npartitions)
+        (name1, i): (collect, p, i, df._meta, barrier_token) for i in range(npartitions)
     }
+    cleanup_token = "cleanup-" + always_new_token
+    dsk5 = {cleanup_token: (cleanup_partd_files, p, list(dsk4))}
```

Comment:
I think that you're bringing back all of the results into a single task with `list(dsk4)`.

If we do this approach then we might have to have an intermediary set of tasks that take in the results of dsk4 and return None for each.  In this way we maintain the dependencies, but don't move the data around.

Reply:
Mmm OK. Do you know, does `dsk3` at https://github.com/dask/dask/pull/5977/files#diff-83ae5352ddc87bd80c831102addd9b1eL407 have a similar problem? Perhaps that result isn't as large.

Reply:
I don't think so, my guess is that the output tasks in dsk2 write things to disk and then return `None`.  

The output tasks of `dsk4` read from disk and return dataframes, so this is more of a concern.

Instead, I think that the solution is to have a `dsk4b` that maps `lambda x: None` across the outputs of dsk4.